docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+understand what it means for them to terminate?
+
+The variable `returnValue` will contain the result of the binary search, when
+the search terminated. Recall, there is no execution stack. Hence, we introduce
+the variable `returnValue` right away. The downside is that we have to do
+book-keeping for this variable.
+
+**Initialize variables.** Having introduced the variables, we have to
+initialize them. That is, we want to specify lines 2-3 of the Java code:
+
+```java
+1:     public static int binarySearch(int[] a, int key) {
+2:         int low = 0;
+3:         int high = a.length - 1;
+           ...
+17:    }
+```
+
+To this end, we change the body of the predicate `Init` to the following:
+
+```tla
+Init ==
+  low = 0 /\ high = Len(INPUT_SEQ) - 1 /\ isTerminated = FALSE /\ returnValue = 0
+```
+
+You probably have guessed, what the above line means. You are probably a bit
+puzzled about the mountain-like operator `/\`. It is called *conjunction*,
+which is usually written as `&&` or `and` in programming languages. The
+important effect of the above expression is that every variable in the
+left-hand side of `=` is required to have the value specified in the right-hand
+side of `=`. It is important to know that TLA+ does not impose any particular
+order of evaluation for `/\`. However, both Apalache and TLC evaluate
+some expressions of the form `x = e` in the initialization predicate as
+assignments. Hence, it is often a good idea to think about `/\` as being
+evaluated from left to right.
+
+As it is hard to fit many expressions in one line, TLA+ offers special syntax
+for writing a big conjunction. Here is the standard way of writing `Init`
+(indentation is important):
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch2.tla:46:51}}
+```
+
+The above lines do not deserve a lot of explanation. As you have guessed,
+`Len(INPUT_SEQ)` computes the length of the input sequence.
+
+**Update variables.** Having done all the preparatory work, we are now ready to
+specify the behavior in lines 5 and 16 of `binarySearch`.
+
+```java
+1:     public static int binarySearch(int[] a, int key) {
+2:         int low = 0;
+3:         int high = a.length - 1;
+4:
+5:         while (low <= high) {
+            ...
+15:        }
+16:        return -(low + 1);  // key not found.
+17:    }
+```
+
+To this end, we redefine `Next` as follows:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch2.tla:53:62}}
+```
+
+Most likely, you have no problem reading this definition, except for the part
+that includes `isTerminated'`, `returnValue'`, and `UNCHANGED`. The expression
+`isTerminated' = TRUE` means that `isTerminated` has the value `TRUE` in the
+next state.  Likewise, `returnValue' = -(low + 1)` means that `returnValue` has
+the value `-(low + 1)` in the next state. The expression `UNCHANGED <<low,
+high>>` is a convenient shortcut for writing `low' = low /\ high' = high`.
```

Comment:
You may want to expand on why having `UNCHANGED` is important. It will not be obvious for the level of reader targeted here.